resources/views/oauth2/profile/edit-client.blade.php

```diff
@@ -1,171 +1,213 @@
+        }
+
+        const ClientTypes = {
+            Public: '{!! OAuth2\Models\IClient::ClientType_Public !!}',
+            Confidential: '{!! OAuth2\Models\IClient::ClientType_Confidential !!}'
+        }
+
+        @foreach($client->admin_users as $admin_user)
+        adminUsers.push({
+            'id': {!! $admin_user->id !!},
+            'fullName': '{!! $admin_user->first_name !!} {!! $admin_user->last_name !!}'
         });
         @endforeach
 
-        var scopes = ['openid'];
-        @if ($client->use_refresh_token)
-        scopes.push('offline_access');
-        @endif
-        @foreach ($scopes as $scope)
-        @if ( in_array($scope->id, $selected_scopes))
-            scopes.push('{!!trim($scope->name)!!}');
-        @endif
+        @foreach($scopes as $scope)
+        scopes.push({
+            'id': {!! $scope->id !!},
+            'name': '{!! $scope->name !!}',
+            'description': '{!! $scope->description !!}',
+            'api_name': '{!! $scope->getApiName() !!}',
+            'api_logo': '{!! $scope->getApiLogo() !!}',
+            'api_description': '{!! $scope->getApiDescription() !!}',
+        });
         @endforeach
 
-        $(document).ready(function () {
-            $('.panel-collapse').collapse('hide');
-            location.hash && $(location.hash + '.collapse').collapse('show');
+        @foreach($selected_scopes as $selected_scope)
+        selectedScopes.push({!! $selected_scope !!});
+        @endforeach
 
-            $(document).on('click', '.head-button', function(e){
-                $('.panel-collapse').collapse('hide');
-                window.location.hash = $(this).attr('href');
-            });
+        @foreach(OAuth2\OAuth2Protocol::getSigningAlgorithmsPerClientType($client) as $alg)
+        supportedSigningAlgorithms.push('{!! $alg !!}');
+        @endforeach
 
-            $(document).on('click', '.copy-scopes-button', function(e){
-                // Copy the text inside the text field
+        @foreach(OAuth2\OAuth2Protocol::getKeyManagementAlgorithmsPerClientType($client) as $alg)
+        supportedKeyManagementAlgorithms.push('{!! $alg !!}');
+        @endforeach
 
-                navigator.clipboard.writeText(scopes.join(' '));
-                // Alert the copied text
-                alert("Copied Scopes: " + scopes.join(' '));
-            });
-        });
+        @foreach(OAuth2\OAuth2Protocol::$supported_content_encryption_algorithms as $alg)
+        supportedContentEncryptionAlgorithms.push('{!! $alg !!}');
+        @endforeach
+
+        @foreach(OAuth2\OAuth2Protocol::getTokenEndpointAuthMethodsPerClientType($client) as $method)
+        supportedTokenEndpointAuthMethods.push('{!! $method !!}');
+        @endforeach
+
+        @foreach(\jwk\JSONWebKeyTypes::$supported_keys as $type)
+        supportedJSONWebKeyTypes.push('{!! $type !!}');
+        @endforeach
+
+        const initialValues = {
```

Comment:
dont do this, use a serializer from controller don not pass the client raw entity